core/src/main/java/io/parsingdata/metal/token/Pre.java

```diff
@@ -57,7 +58,7 @@ public Pre(final String name, final Token token, final Expression predicate, fin
             return failure();
         }
         final Optional<Environment> result = token.parse(scope, environment.addBranch(this), encoding);
-        return result.isPresent() ? success(result.get().closeBranch()) : failure();
+        return result.map(resultEnvironment -> success(resultEnvironment.closeBranch())).orElseGet(Util::failure);
```

Comment:
Code style: In `Rep` (below), these span multiple lines, let's do that everywhere. So:

```java
return result
    .map(resultEnvironment -> success(resultEnvironment.closeBranch()))
    .orElseGet(Util::failure);
```